radicle-httpd/src/lib.rs

```diff
@@ -100,13 +100,16 @@ async fn git_handler(
     query: RawQuery,
     body: Bytes,
 ) -> impl IntoResponse {
+    let mut response_headers = HeaderMap::new();
     let query = query.0.unwrap_or_default();
-    let id: Id = project.strip_suffix(".git").unwrap_or(&project).parse()?;
+    let id: Id = match project.strip_suffix(".git").unwrap_or(&project).parse() {
+        Ok(id) => id,
+        Err(_) => return Ok::<_, Error>((StatusCode::NOT_FOUND, response_headers, vec![])),
```

Comment:
Is this not a user error instead of not found? :thinking: Seems like a `400` since the repository identifier can't be parsed.

Reply:
Hmm that would work too, all 4** errors are user errors -- the main thing is to not return a 500 here.

Reply:
The thing is though, that it's not clear that the user is requesting a git project. The way we determine that is by checking if we can parse. So I think 404 is ok here, because it could be that the user just hit the wrong api route.

Reply:
Yeah, I think 404 is more appropriate here as well:
- the `Id` was not found
- the route was not found